nbgrader_to_canvas/__init__.py

```diff
@@ -0,0 +1,91 @@
+import os
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.middleware.proxy_fix import ProxyFix
+from flask_sqlalchemy import SQLAlchemy
+from prometheus_flask_exporter import PrometheusMetrics
+from flask_session import Session
+from circuitbreaker import circuit
+
+from . import settings
+
+__version__ = '0.0.1'
+
+# initialize the app
+app = Flask(__name__, template_folder='./templates')
+
+# filesystem session interface, see https://github.com/fengsp/flask-session/blob/master/docs/index.rst#built-in-session-interfaces
+#app.config['PERMANENT_SESSION_LIFETIME'] = timedelta(hours=5)
+#app.config['SESSION_FILE_DIR'] = '/tmp'
+#app.config['SESSION_PERMANENT'] = True
+# The maximum number of items the session stores 
+# before it starts deleting some, default 500
+#app.config['SESSION_FILE_THRESHOLD'] = 100  
+
+# db session interface instead of filesystem due to user tokens stored in db Users table
+# see sqlalchemy code after init_app below too
+app.config['SQLALCHEMY_DATABASE_URI'] = os.getenv('DATABASE_URI')
+app.config['SESSION_TYPE'] = 'sqlalchemy'
+app.config['SQLALCHEMY_POOL_TIMEOUT'] = 600
+app.config['SQLALCHEMY_POOL_RECYCLE'] = 300
+
+app.secret_key = settings.secret_key
+app.config.from_object(settings.configClass)
+
+#app.config['SECRET_KEY'] = config.SECRET_KEY
+
+# add middleware
+app.wsgi_app = ProxyFix(app.wsgi_app)
+
+
+#
+# intialize all other modules here due to circ. import errors
+#
+
+# initialize db
+
```

Comment:
might want to hide this behind a repository class. i generally advocate for a layer of abstraction between the "business logic" and the database driver.